lib/collections.go

```diff
@@ -948,6 +1020,44 @@ func zipLists(arg0, arg1 ref.Val) ref.Val {
 	return types.NewRefValMap(types.DefaultTypeAdapter, m)
 }
 
+func mapKeys(val ref.Val) ref.Val {
+	mapK, ok := val.(traits.Mapper)
+	if !ok {
+		return types.ValOrErr(mapK, "no such overload")
+	}
+	n, _ := mapK.Size().(types.Int)
+	new := make([]ref.Val, 0, n)
+	if mapK.Size() != types.IntZero {
+		m, err := mapK.ConvertToNative(refValMap)
```

Comment:
Note that we can iterate directly since `mapK` is a `traits.Iterable`. In cases where the values are large, this would be better since they do not need to be converted.